modules/elasticsearch_helper_index_management/src/ElasticsearchQueueManager.php

```diff
@@ -0,0 +1,202 @@
+   *
+   * @var \Drupal\elasticsearch_helper\Plugin\ElasticsearchIndexManager
+   */
+  protected $pluginManagerElasticsearchIndexProcessor;
+
+  /**
+   * Drupal\Core\Entity\EntityTypeManagerInterface definition.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Database table name.
+   *
+   * @var string
+   */
+  protected $table = 'es_reindex_queue';
+
+  /**
+   * Constructs a new ElasticsearchQueueManager object.
+   */
+  public function __construct(Connection $database, ElasticsearchIndexManager $plugin_manager_elasticsearch_index_processor, EntityTypeManagerInterface $entity_type_manager) {
+    $this->database = $database;
+    $this->pluginManagerElasticsearchIndexProcessor = $plugin_manager_elasticsearch_index_processor;
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getStatus($index_id) {
+    return [
+      'total' => $this->getTotal($index_id),
+      'processed' => $this->getProcessed($index_id),
+      'errors' => $this->getTotalErrors($index_id),
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getTotal($index_id) {
+    $definition = $this->pluginManagerElasticsearchIndexProcessor->getDefinition($index_id);
+
+    $query = $this->database->select($this->table);
+    $query->condition('plugin_id', $definition['id']);
+    $query->condition('entity_type', $definition['entityType']);
+
+    return $query->countQuery()->execute()->fetchField();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getProcessed($index_id) {
+    $definition = $this->pluginManagerElasticsearchIndexProcessor->getDefinition($index_id);
+
+    $query = $this->database->select($this->table);
+    $query->condition('plugin_id', $definition['id']);
+    $query->condition('entity_type', $definition['entityType']);
+    $query->condition('status', 1);
+
+    return $query->countQuery()->execute()->fetchField();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getTotalErrors($index_id) {
+    $definition = $this->pluginManagerElasticsearchIndexProcessor->getDefinition($index_id);
+
+    $query = $this->database->select($this->table);
+    $query->condition('plugin_id', $definition['id']);
+    $query->condition('entity_type', $definition['entityType']);
+    $query->condition('error', 1);
+
+    return $query->countQuery()->execute()->fetchField();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function addAll($index_id) {
+    $definition = $this->pluginManagerElasticsearchIndexProcessor->getDefinition($index_id);
+
+    $storage = $this->entityTypeManager->getStorage($definition['entityType']);
+
+    if (isset($definition['bundle'])) {
+      $entities = $storage->loadByProperties(['bundle' => [$definition['bundle']]]);
+    }
+    else {
+      $entities = $storage->loadMultiple();
+    }
+
+    foreach ($entities as $entity) {
```

Comment:
Loading the full entities is very heavy process when only the entity id is needed, using entity query instead with $storage->getQuery() would be a lot quicker.

Reply:
Fixed